test/fw/ptl/lib/pbs_testlib.py

```diff
@@ -5560,11 +5571,22 @@ def revert_to_defaults(self, reverthooks=True, revertqueues=True,
             self.manager(MGR_CMD_CREATE, QUEUE, a, id='workq', expect=True)
             setdict.update({ATTR_dfltque: 'workq'})
         if reverthooks:
-            hooks = self.status(HOOK, level=logging.DEBUG)
-            hooks = [h['id'] for h in hooks]
-            a = {ATTR_enable: 'false'}
-            if len(hooks) > 0:
-                self.manager(MGR_CMD_SET, MGR_OBJ_HOOK, a, hooks, expect=True)
+            if self.platform == 'cray' or self.platform == 'craysim':
+                if self.du.cmp(self.hostname, self.dflt_mpp_hook,
+                               self.mpp_hook) != 0:
+                    self.du.run_copy(self.hostname, self.dflt_mpp_hook,
+                                     self.mpp_hook, mode=0644, sudo=True)
+                    self.signal('-HUP')
+                a = {'enabled': 'true'}
+                self.manager(MGR_CMD_SET, MGR_OBJ_PBS_HOOK, a,
+                             'PBS_translate_mpp')
+            else:
+                hooks = self.status(HOOK, level=logging.DEBUG)
```

Comment:
I think this part should be outside of if else and under reverthooks part because we have revert site hooks in cray also. right?